### PR TITLE
Add some more docs on Best Known Configuration

### PR DESCRIPTION
**Pull request by hughsie**
Fixes some of https://github.com/fwupd/fwupd/discussions/5641

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [ ] Feature
- [X] Documentation

### CODE REVIEW

File: docs/best-known-configuration.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
+# Best Known Configuration
+
+Component <tag>s are used both by OEMs and by customers to identify a *known-working* (or
```
Comment (superm1) [inline docs/best-known-configuration.md:3]:
```suggestion
`<tag>`s is used in components used both by OEMs and by customers to identify a *known-working* (or
```

**Review by superm1 — COMMENTED** [review]
The first section needs to be grouped by something to indicate it's for the concept, the second for LVFS and third for local (I only see heading for local)

File: docs/best-known-configuration.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+..and can be synced on the command line:
+
+    $ fwupdmgr sync-bkc
+    ╔══════════════════════════════════════════════════════════════════════════════╗
+    ║ Downgrade System Firmware from 225.52.1521 to 225.53.1649?                   ║
+    ╠══════════════════════════════════════════════════════════════════════════════╣
+    ║ This release sets the number of HBAs supported by the system to 1024.        ║
+    ╚══════════════════════════════════════════════════════════════════════════════╝
+    Perform operation? [Y|n]:
```
Comment (superm1) [inline docs/best-known-configuration.md:96]: this should be in a conceptual section in section 1

**Comment by hughsie** [comment]
> The first section needs to be grouped by something

I've rearranged things a bit, how does that look?

File: docs/best-known-configuration.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+firmwares.
+This would allow an OEM to say *this set of firmware has been tested as a set for workload A,
+and this other set of firmware has been tested for workload B* which is fairly typical for
+enterprise deployments.
+
+## LVFS
+
+The LVFS added support for “vendor-defined” component `<tag>`s in 2021, which are also supported in
+fwupd since version 1.7.3.
+These tags are typically used by server vendors to identify a manifest of firmware for a specific
```
Comment (superm1) [inline docs/best-known-configuration.md:25]: still don't think this should have server in it

File: docs/best-known-configuration.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+## Client
+
+When provisioning the client machine, we can set the BKC by setting `HostBkc=vendor-2021q1` in
+`/etc/fwupd/daemon.conf`.
+Then any invocation of `fwupdmgr sync-bkc` will install or downgrade firmware on all compatible
+devices (UEFI, RAID, network adapter, & SAS HBA etc.) to make the system match a compatible set.
+
+Updating or downgrading firmware away from the *Best Known Configuration* is allowed, but the UI
+shows a warning.
+Using `fwupdmgr sync-bkc` will undo any manual changes and bring the machine back to the BKC.
```
Comment (superm1) [inline docs/best-known-configuration.md:38]: I feel like this should be the last section

File: docs/best-known-configuration.md (hunk 3 above)
Comment (hughsie) [reply docs/best-known-configuration.md:25]: Oopa, removed.

File: docs/best-known-configuration.md (hunk 4 above)
Comment (hughsie) [reply docs/best-known-configuration.md:38]: Maybe Local metadata should be H3? I think it's more common that you set a HostBkc in daemon.conf than define local metadata -- hence why i put it in this order.

Comment (superm1) [reply docs/best-known-configuration.md:38]: Got it, makes sense then.

**Review by superm1 — APPROVED** [review]

**Pull request merged**